django/core/view_helpers.py

```diff
@@ -33,24 +78,48 @@ def get_search_queryset(
         operator = 'and'
     """
 
+    """ 
+    Build text search query
+    """
     if query:
         Query.get(query).add_hit()
-        filters, query = parse_query_string(query, operator="or")
-        criteria.update(filters)
+
+        # this can be used to create split query and filters from search field input text:
+        # `some search terms peer_reviewed:True` -> query="some search terms" and filters=["peer_reviewed": True]
+        # from wagtail.search.utils import parse_query_string
+        # filters, query = parse_query_string(query, operator="and")
+        # criteria.update(filters)
+
+        query = build_search_query(query)
     else:
         query = MATCH_ALL
 
+    # FIXME: tags should be better filters
     for tag in tags:
         query = query & Phrase(tag)
 
+    """ 
+    Filter queryset 
+    """
     if criteria:
+        logger.debug(f"criteria={criteria}")
         try:
             queryset = queryset.filter(**criteria)
-        except FieldError:
-            logger.warning("Invalid filter criteria: %s", criteria)
+        except FieldError as e:
+            logger.warning(f"Invalid filter criteria: {criteria}. Error: {str(e)}")
```

Comment:
use `exc_info=e` instead of converting the exception into a string and missing out on the entire stack trace